CanvasPlusPlayground/Views/CourseGradeView.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  CourseGradeView.swift
+//  CanvasPlusPlayground
+//
+//  Created by Songyuan Liu on 9/13/24.
+//
+
+import SwiftUI
+
+struct CourseGradeView: View {
+    let course: Course
+    @State private var enrollment: Enrollment?
+    @State private var gradeManager: CourseGradeManager
+    
+    init(course: Course) {
+        self.course = course
+        _gradeManager = .init(initialValue: CourseGradeManager())
+    }
+
+
+    var body: some View {
+        List {
+            if let score = enrollment?.grades?.currentScore {
+                Text("Current Score: \(score)")
+            } else {
+                Text("Current Score unavailable")
+            }
+            
+            if let grade = enrollment?.grades?.currentGrade {
+                Text("Current Grade: \(grade)")
+            } else {
+                Text("Current grade unavailable")
+            }
+            
+            if let finalScore = enrollment?.grades?.finalScore {
+                Text("Final Score: \(finalScore)")
+            } else {
+                Text("Final Score unavailable")
+            }
+            
+            if let finalGrade = enrollment?.grades?.finalGrade {
+                Text("Current Score: \(finalGrade)")
+            } else {
+                Text("Final Grade unavailable")
+            }
+           
+        }
+        .task {
+            await gradeManager.fetchEnrollments()
```

Comment:
@ecoist-ste Is it possible that we fetch enrollments at the `CourseListView`? Then we'd have less repetitive work - I'm assuming the enrollments are the same as courses.

Reply:
That sounds good. I will do that instead